diff --git a/php7/memcache.c b/php7/memcache.c
index 89ca4e0..631a07f 100644
--- a/php7/memcache.c
+++ b/php7/memcache.c
@@ -281,26 +281,26 @@ static PHP_INI_MH(OnUpdatePrefixStaticKey) /* {{{ */
 
 /* {{{ PHP_INI */
 PHP_INI_BEGIN()
-	STD_PHP_INI_ENTRY("memcache.allow_failover",		"1",			PHP_INI_ALL, OnUpdateLong,			allow_failover,	zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.max_failover_attempts",	"20",			PHP_INI_ALL, OnUpdateFailoverAttempts,		max_failover_attempts,	zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.default_port",			"11211",		PHP_INI_ALL, OnUpdateLong,			default_port,	zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.chunk_size",			"32768",		PHP_INI_ALL, OnUpdateChunkSize,		chunk_size,		zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.protocol",				"ascii",		PHP_INI_ALL, OnUpdateProtocol,		protocol,		zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.hash_strategy",			"consistent",	PHP_INI_ALL, OnUpdateHashStrategy,	hash_strategy,	zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.hash_function",			"crc32",		PHP_INI_ALL, OnUpdateHashFunction,	hash_function,	zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.redundancy",			"1",			PHP_INI_ALL, OnUpdateRedundancy,	redundancy,			zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.session_redundancy",	"2",			PHP_INI_ALL, OnUpdateRedundancy,	session_redundancy,	zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.compress_threshold",	"20000",		PHP_INI_ALL, OnUpdateCompressThreshold,	compress_threshold,	zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.lock_timeout",			"15",			PHP_INI_ALL, OnUpdateLockTimeout,		lock_timeout,		zend_memcache_globals,	memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.session_prefix_host_key",       			"0",			PHP_INI_ALL, OnUpdateBool, session_prefix_host_key, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.session_prefix_host_key_remove_www",    	"1",			PHP_INI_ALL, OnUpdateBool, session_prefix_host_key_remove_www, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.session_prefix_host_key_remove_subdomain",  "0",			PHP_INI_ALL, OnUpdateBool, session_prefix_host_key_remove_subdomain, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.session_prefix_static_key",         		NULL,			PHP_INI_ALL, OnUpdatePrefixStaticKey, session_prefix_static_key, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.session_save_path",         				NULL,			PHP_INI_ALL, OnUpdateString, session_save_path, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.prefix_host_key",       					"0",			PHP_INI_ALL, OnUpdateBool, prefix_host_key, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.prefix_host_key_remove_www",    			"1",			PHP_INI_ALL, OnUpdateBool, prefix_host_key_remove_www, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.prefix_host_key_remove_subdomain",  		"0",			PHP_INI_ALL, OnUpdateBool, prefix_host_key_remove_subdomain, zend_memcache_globals, memcache_globals)
-	STD_PHP_INI_ENTRY("memcache.prefix_static_key",         				NULL,			PHP_INI_ALL, OnUpdatePrefixStaticKey, prefix_static_key, zend_memcache_globals, memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.allow_failover",								"1",			PHP_INI_ALL, OnUpdateBool,				allow_failover,								zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.max_failover_attempts",						"20",			PHP_INI_ALL, OnUpdateFailoverAttempts,	max_failover_attempts,						zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.default_port",								"11211",		PHP_INI_ALL, OnUpdateLong,				default_port,								zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.chunk_size",									"32768",		PHP_INI_ALL, OnUpdateChunkSize,			chunk_size,									zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.protocol",									"ascii",		PHP_INI_ALL, OnUpdateProtocol,			protocol,									zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.hash_strategy",								"consistent",	PHP_INI_ALL, OnUpdateHashStrategy,		hash_strategy,								zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.hash_function",								"crc32",		PHP_INI_ALL, OnUpdateHashFunction,		hash_function,								zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.redundancy",									"1",			PHP_INI_ALL, OnUpdateRedundancy,		redundancy,									zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.session_redundancy",							"2",			PHP_INI_ALL, OnUpdateRedundancy,		session_redundancy,							zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.compress_threshold",							"20000",		PHP_INI_ALL, OnUpdateCompressThreshold,	compress_threshold,							zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.lock_timeout",								"15",			PHP_INI_ALL, OnUpdateLockTimeout,		lock_timeout,								zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.session_prefix_host_key",       				"0",			PHP_INI_ALL, OnUpdateBool,				session_prefix_host_key,					zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.session_prefix_host_key_remove_www",    		"1",			PHP_INI_ALL, OnUpdateBool,				session_prefix_host_key_remove_www,			zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.session_prefix_host_key_remove_subdomain",	"0",			PHP_INI_ALL, OnUpdateBool,				session_prefix_host_key_remove_subdomain,	zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.session_prefix_static_key",         			NULL,			PHP_INI_ALL, OnUpdatePrefixStaticKey,	session_prefix_static_key,					zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.session_save_path",         					NULL,			PHP_INI_ALL, OnUpdateString,			session_save_path,							zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.prefix_host_key",       						"0",			PHP_INI_ALL, OnUpdateBool,				prefix_host_key,							zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.prefix_host_key_remove_www",    				"1",			PHP_INI_ALL, OnUpdateBool,				prefix_host_key_remove_www,					zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_BOOLEAN("memcache.prefix_host_key_remove_subdomain",  			"0",			PHP_INI_ALL, OnUpdateBool,				prefix_host_key_remove_subdomain,			zend_memcache_globals,	memcache_globals)
+	STD_PHP_INI_ENTRY(	"memcache.prefix_static_key",         					NULL,			PHP_INI_ALL, OnUpdatePrefixStaticKey,	prefix_static_key,							zend_memcache_globals,	memcache_globals)
 PHP_INI_END()
 /* }}} */
 
@@ -519,7 +519,7 @@ PHP_MINFO_FUNCTION(memcache)
 	php_info_print_table_start();
 	php_info_print_table_header(2, "memcache support", "enabled");
 	php_info_print_table_row(2, "Version", PHP_MEMCACHE_VERSION);
-	php_info_print_table_row(2, "Revision", "$Revision$");
+	php_info_print_table_row(2, "Revision", "NON_BLOCKING_IO_php7 - baec8a2");
 	php_info_print_table_end();
 
 	DISPLAY_INI_ENTRIES();
diff --git a/php7/memcache_pool.h b/php7/memcache_pool.h
index 2382aa8..105d5f5 100644
--- a/php7/memcache_pool.h
+++ b/php7/memcache_pool.h
@@ -408,22 +408,22 @@ ZEND_BEGIN_MODULE_GLOBALS(memcache)
 	long protocol;
 	long hash_strategy;
 	long hash_function;
-	long allow_failover;
+	zend_bool allow_failover;
 	long max_failover_attempts;
 	long redundancy;
 	long session_redundancy;
 	long compress_threshold;
 	long lock_timeout;
 	char *session_key_prefix;
-	char *session_prefix_host_key;
-	char *session_prefix_host_key_remove_www;
-	char *session_prefix_host_key_remove_subdomain;
+	zend_bool session_prefix_host_key;
+	zend_bool session_prefix_host_key_remove_www;
+	zend_bool session_prefix_host_key_remove_subdomain;
 	char *session_prefix_static_key;
 	char *session_save_path;
 	char *key_prefix;
-	char *prefix_host_key;
-	char *prefix_host_key_remove_www;
-	char *prefix_host_key_remove_subdomain;
+	zend_bool prefix_host_key;
+	zend_bool prefix_host_key_remove_www;
+	zend_bool prefix_host_key_remove_subdomain;
 	char *prefix_static_key;
 ZEND_END_MODULE_GLOBALS(memcache)
 
